src/block_cache/block_chunkcache.c

```diff
@@ -249,6 +249,87 @@ __chunkcache_eviction_thread(void *arg)
     return (WT_THREAD_RET_VALUE);
 }
 
+/*
+ * __chunkcache_str_cmp --
+ *     Qsort function: sort string array.
+ */
+static int WT_CDECL
+__chunkcache_str_cmp(const void *a, const void *b)
+{
+    return (strcmp(*(const char **)a, *(const char **)b));
```

Comment:
1. [This](https://github.com/wiredtiger/wiredtiger/pull/1821#issuecomment-86302940) should explain the reason.
2. The `void *` here is the address to the `char *`, therefore, I had to first typecast it to `char**` and then dereference it. Directly dereferencing a `void *` will result in compilation errors. 

> Can we just pass `strcmp` to the qsort call?

For the same reason above, I believe we cannot directly pass `strcmp` to `qsort`.

3. I picked this comment pattern from different places using Qsort. For example -

```
/*
 * __evict_lru_cmp --
 *     Qsort function: sort the eviction array.
 */
static int WT_CDECL
__evict_lru_cmp(const void *a_arg, const void *b_arg)
{
```

Reply:
Hmm I still don't get the `WT_CDECL` call thing sorry. Why do we need it on this strcmp function and not anywhere else? What happens if it's not there?

For the cast and `strcmp` questions - those answers make sense, thanks for that.

Reply:
`__cdecl` specifies a certain "calling convention".

The `WT_CDECL` macro is set up in two different files - o`s_windows.h` and `posix.h`. In the `os_windows.h` file, it's set as `__cdecl`, which is a rule that tells the compiler how to handle function calls. This rule is commonly used in Windows systems (Microsoft compiler).

The standard C function (qsort) have a compare function like this signature:

`int ( __cdecl *compare ) (const void *key, const void *datum)`

On the other hand, in the `posix.h` file, `WT_CDECL` isn't assigned anything. This is because, in `POSIX` systems don't need to specify this rule. The system's compiler automatically takes care of the details for function calls.

Reply:
You would see it being used for all the compare functions in our code.

Reply:
OK I did a little digging and I think I see what's going on here - we need this for all C standard library functions we provide callbacks to, so `qsort` and `bsearch`. I'm OK with leaving it in - we theoretically no longer need it now that we've got our own `qsort` implementation, but `bsearch` still needs it and there's no harm leaving it.